### ISSUE TITLE
Rule: class_delegate_protocol only checks for suffix "delegate" on inherited protocol, not if it also conforms to `class`

### ISSUE THREAD

**Issue by RollingGoron** [realm/SwiftLint#1994]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

example

Swiftlint will throw a warning stating that it needs class keyword added, even though the inherited one already does. If you do add class the Swift compiler complains about redundant keyword being used. After browsing the code, it's not checking if the inherited protocol also includes the class keyword.

```swift
protocol MyCoolProtocol: class {
  func doCoolThing()
}

protocol MyAwesomeDelegate: MyCoolProtocol { //Swiftlint throws an warning here about `class_delegate_protocol`
  func doCoolThing()
  func doAwesomeThing()
}
```

**Comment by marcelofabri** [comment]
Duplicate of #1939.

Currently, SwiftLint only knows about the file that it's being analyzed at the moment. It would be possible to catch the case you've described, but it would fail as soon as `MyCoolProtocol` is in another file.
